components/cascader/index.tsx

```diff
@@ -165,22 +165,20 @@ export default class Cascader extends React.Component<CascaderProps, CascaderSta
       inputValue: '',
       inputFocused: false,
       popupVisible: props.popupVisible,
-      flattenOptions:
-        props.showSearch ? this.flattenTree(props.options, props.changeOnSelect, props.fieldNames) : undefined,
     };
   }
 
-  componentWillReceiveProps(nextProps: CascaderProps) {
-    if ('value' in nextProps) {
-      this.setState({ value: nextProps.value || [] });
+  componentDidUpdate(prevProps: CascaderProps) {
```

Comment:
Do not use `componentDidUpdate` instead of `componentWillReceiveProps`. Use `getDerivedStateFromProps` instead.

Reply:
Why componentDidUpdate isn't the right place for this code?

Reply:
`getDerivedStateFromProps` is the replacement function of `componentWillReceiveProps`. It's better use it here if you don't have effect code.